src/ReTestItems.jl

```diff
@@ -337,21 +350,51 @@ function _runtests_in_current_env(
     return nothing
 end
 
-function start_worker(proj_name, nworker_threads, worker_init_expr, ntestitems)
+# Start a new `Worker` with `nworker_threads` threads and evaluate `worker_init_expr` on it.
+# The provided `worker_num` is only for logging purposes, and not persisted as part of the worker.
+function start_worker(proj_name, nworker_threads, worker_init_expr, ntestitems; worker_num=nothing)
     w = Worker(; threads="$nworker_threads")
+    i = worker_num == nothing ? "" : " $worker_num"
     # remote_fetch here because we want to make sure the worker is all setup before starting to eval testitems
     remote_fetch(w, quote
         using ReTestItems, Test
         Test.TESTSET_PRINT_ENABLE[] = false
         const GLOBAL_TEST_CONTEXT = ReTestItems.TestContext($proj_name, $ntestitems)
         GLOBAL_TEST_CONTEXT.setups_evaled = ReTestItems.TestSetupModules()
-        @info "Starting test item evaluations on pid = $(Libc.getpid()), with $(Threads.nthreads()) threads"
+        @info "Starting test worker$($i) on pid = $(Libc.getpid()), with $(Threads.nthreads()) threads"
         $(worker_init_expr.args...)
         nothing
     end)
     return w
```

Comment:
My theory is that there is a race between the internal tasks we spawn in `Worker` and the lethal `worker_init_expr` here. So I tried to running the expression in a separate `remote_fetch`, after the we know everything has been setup for the previous `remote_fetch`, and now I'm running `"worker crashes immediately"` test in a while loop for a couple of minutes without a hang.
```suggestion
        nothing
    end)
    remote_fetch(w, worker_init_expr)
    return w
```

Reply:
Nope, I was not interpolating the expression correctly... sorry for the noise